app/workers/builds.php

```diff
@@ -0,0 +1,262 @@
+        \curl_setopt($ch, CURLOPT_TIMEOUT, 900);
+        \curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+        \curl_setopt($ch, CURLOPT_HTTPHEADER, [
+            'Content-Type: application/json',
+            'x-appwrite-project: '.$projectId,
+            'x-appwrite-executor-key: '. App::getEnv('_APP_EXECUTOR_SECRET', '')
+        ]);
+
+        $response = \curl_exec($ch);
+        $responseStatus = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+        $error = \curl_error($ch);
+        if (!empty($error)) {
+            throw new \Exception($error);
+        }
+
+        \curl_close($ch);
+
+        if ($responseStatus !== 200) {
+            throw new \Exception("Build failed with status code: $responseStatus");
+        }
+
+        $response = json_decode($response, true);
+        if (isset($response['error'])) {
+            throw new \Exception($response['error']);
+        }
+
+        if (isset($response['success']) && $response['success'] === true) {
+            return;
+        } else {
+            throw new \Exception("Build failed");
+        }
+    }
+
+    protected function triggerCreateRuntimeServer(string $projectId, string $functionId, string $tagId) 
+    {
+        $ch = \curl_init();
+        \curl_setopt($ch, CURLOPT_URL, "http://appwrite-executor:8080/v1/create/runtime");
+        \curl_setopt($ch, CURLOPT_POST, true);
+        \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        \curl_setopt($ch, CURLOPT_TIMEOUT, 900);
+        \curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+        \curl_setopt($ch, CURLOPT_HTTPHEADER, [
+            'Content-Type: application/json',
+            'x-appwrite-project: '.$projectId,
+            'x-appwrite-executor-key: '. App::getEnv('_APP_EXECUTOR_SECRET', '')
+        ]);
+        \curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode([
+            'functionId' => $functionId,
+            'tagId' => $tagId
+        ]));
+
+        $response = \curl_exec($ch);
+        $responseStatus = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+        $error = \curl_error($ch);
+        if (!empty($error)) {
+            throw new \Exception($error);
+        }
+
+        \curl_close($ch);
+
+        if ($responseStatus !== 200) {
+            throw new \Exception("Build failed with status code: $responseStatus");
+        }
+
+        $response = json_decode($response, true);
+        if (isset($response['error'])) {
+            throw new \Exception($response['error']);
+        }
+
+        if (isset($response['success']) && $response['success'] === true) {
+            return;
+        } else {
+            throw new \Exception("Build failed");
+        }
+    }
+
+    protected function buildTag(string $projectId, string $functionId, string $tagId) 
+    {
+        $dbForProject = $this->getProjectDB($projectId);
+        
+        // TODO: Why does it need to skip authorization?
+        $function = Authorization::skip(fn() => $dbForProject->getDocument('functions', $functionId));
```

Comment:
For some reason, `getDocument()` returns an empty object when I make the call without the `::skip` method. 
I need some eyes here 👀